aten/src/ATen/native/cuda/Distributions.cpp

```diff
@@ -0,0 +1,72 @@
+#include <ATen/native/cuda/Distributions.h>
+#include <ATen/Functions.h>
+#include <ATen/NativeFunctions.h>
+#include <ATen/TensorIterator.h>
+#include <ATen/CUDAGeneratorImpl.h>
+
+namespace at { namespace native {
+
+Tensor _s_poisson_cuda(const Tensor& lambda, c10::optional<Generator> gen_) {
+  auto gen = get_generator_or_default<CUDAGeneratorImpl>(gen_, cuda::detail::getDefaultCUDAGenerator());
```

Comment:
these calls are flagging:
> Call to at::cuda::detail::getDefaultCUDAGenerator is logically private, please avoid use outside of its parent namespace

Do you think this is bogus or is there any refactoring that makes sense here?

Reply:
It sounds like it doesn't like that a `detail` namespace is being used outside of the `at::cuda` namespace but that's just how ATen is structured.